src/dictionaries.cpp

```diff
@@ -0,0 +1,42 @@
+#include "dictionaries.h"
+#include <utility>
+
+BasicDictionary::BasicDictionary()
+{
+}
+
+std::string BasicDictionary::getName(uint8_t type) const
+{
+    return m_rightDict.at(type);
+}
+
+uint8_t BasicDictionary::getType(const std::string& name) const
+{
+    return m_reverseDict.at(name);
+}
+
+void BasicDictionary::add(uint8_t type, const std::string& name)
+{
+    m_rightDict.emplace(std::make_pair(type, name));
```

Comment:
With `std::map::emplace` you don't need to construct pairs manually:
`m_rightDict.emplace(type, name);`